pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -140,3 +144,9 @@ func NewAuthInfo() *AuthInfo {
 func NewPreferences() *Preferences {
 	return &Preferences{Extensions: make(map[string]runtime.Object)}
 }
+
+// AuthProviderConfig
+type AuthProviderConfig struct {
+	Name   string           `json:"name"`
+	Config *json.RawMessage `json:"config"`
```

Comment:
> I'm still looking for where (if) the extensions are actually registered/used...

They were there so that deep integrators like openshift would be able to supply custom configuration values and the kube decoding/encoding path wouldn't strip them out.  They aren't natively used in kube.

Since the core kube code will actually be interpreting these values, the structure you've described here is more suitable.  If you were going to say, "we aren't going to provide this, build your own "gcp" on top", then you'd point them to the existing `Extensions` field and let them do their thing.